### PR TITLE
Refactoring code using the older SchemaTemplate object to now use the…

### PR DESCRIPTION
**Pull request by maniarathi**
… NewSchemaTemplate. This does not delete the older SchemaTemplate file yet.

### CODE REVIEW

File: ingest/importer/conversion/template_manager.py [hunk 1]
```diff
@@ -130,28 +130,26 @@ def get_worksheet_domain_type(self, title):
 
     def get_key_for_label(self, header_name, tab_name):
```
[the remark this answers is not in the document]
Comment (maniarathi) [reply ingest/importer/conversion/template_manager.py]: Done!

File: ingest/template/new_schema_template.py [hunk 2]
```diff
@@ -121,6 +125,33 @@ def lookup_property_attributes_in_metadata(self, property_key):
...
+        :return: A string representing the display name's fully qualified path from a schema.
+        """
+
+        try:
+            return self.tab_config.get_key_for_label(tab_display_name)
+        except KeyError:
+            raise UnknownKeySchemaException(
+                f"ERROR: Was unable to find a fully qualified path (key) with the display name {tab_display_name}.")
+
+    def lookup_fully_qualified_property_path_for_column_name_in_tab(self, column_name, tab_display_name):
```
[the remark this answers is not in the document]
Comment (maniarathi) [reply ingest/template/new_schema_template.py]: Good catch!

File: ingest/template/new_schema_template.py [hunk 3]
```diff
@@ -121,6 +125,33 @@ def lookup_property_attributes_in_metadata(self, property_key):
 
         return self._lookup_fully_qualified_key_path_in_dictionary(property_key, self.meta_data_properties)
 
+    def lookup_fully_qualified_property_path_given_tab_display_name(self, tab_display_name):
```
[the remark this answers is not in the document]
Comment (maniarathi) [reply ingest/template/new_schema_template.py]: Great ideas! I changed it to `lookup_metadata_schema_name_given_title`.

File: ingest/template/descriptor.py [hunk 4]
```diff
@@ -60,7 +62,7 @@ def __init__(self, json_data):
             self.value_type = json_data["items"]["type"]
 
         self.format = json_data.get("format")
-        self.external_reference = False
+        self.external_reference = json_data.get("external_reference", False)
```
[the remark this answers is not in the document]
Comment (maniarathi) [reply ingest/template/descriptor.py]: Ahh ok. It looks like identifiable properties are the ones that are external references so I added that logic back in. Thanks!

File: ingest/template/descriptor.py [hunk 5]
```diff
@@ -116,6 +117,7 @@ def __init__(self, json_data):
                 # identifiable properties
                 if property_name in IDENTIFIABLE_PROPERTIES:
                     child_property_descriptor.identifiable = True
+                    child_property_descriptor.external_reference = True
```
Comment (aaclan-ebi) [inline ingest/template/descriptor.py]: The logic is actually, if property name is `uuid` , it'll be considered as external reference. So in the spreadsheet, you can have donor.uuid, cell_suspension.uuid, sequencing_file.uuid then that means it's referring to an entity already submitted before. It's used for linking to an existing entity or when specifying the updates in the spreadsheet.

Comment (maniarathi) [reply ingest/template/descriptor.py]: Done! Thank you for the explanation!

**Pull request merged**